packages/core/src/odata-common/selectable/field.ts

```diff
@@ -23,6 +23,29 @@ export type FieldType =
  */
 export type DeepFieldType = FieldType | { [keys: string]: DeepFieldType };
 
+export interface FieldOptions<
```

Comment:
why is it necessary to have the generic here? Simple booleans enough?

Reply:
No, because both those information are relevant to types down the line. E.g., when looking at the filter, I want to have something like `equals(string | null)`. A boolean does not allow me to do any type magic, because its type is only determined at runtime. It is `true` **and** `false` at design time.